### PR TITLE
Add Schedule, Event and AddEventCommand

### PR DESCRIPTION
**Pull request by tyanhan**

### CODE REVIEW

File: src/main/java/seedu/address/model/schedule/Schedule.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    public List<Event> getEvents() {
+        return Collections.unmodifiableList(events);
+    }
+
+    public void addEvent(Event event) {
+        events.add(event);
+    }
+
+    public void setEvents(List<Event> events) {
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/model/schedule/Schedule.java:29]: Maybe include javadocs for this method. Or change the name of the method. Not quite sure what this does.

File: src/main/java/seedu/address/model/schedule/Event.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * Represents a scheduled Event.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Event {
+
+    private final EventDescription eventDescription;
+    private final LocalDate date;
+    private final LocalTime time;
+    private final Duration duration;
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/model/schedule/Event.java:16]: Is there any benefit to using the Duration library as compared to just int or double? Is there more compatibility to LocalTime using Duration?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -95,6 +100,69 @@ public static Email parseEmail(String email) throws ParseException {
...
+     * Parses a {@code String duration} into an {@code Duration}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code duration} is invalid.
+     */
+    public static Duration parseDuration(String duration) throws ParseException {
+        requireNonNull(duration);
+        String trimmedDuration = duration.trim();
+        try {
+            return Duration.ofHours(Integer.parseInt(trimmedDuration));
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:160]: What if the duration is 90 minutes long?

File: src/main/java/seedu/address/model/schedule/Event.java (hunk 2 above)
Comment (tyanhan) [reply src/main/java/seedu/address/model/schedule/Event.java:16]: Yup, you can refer to this https://www.baeldung.com/java-period-duration for more information about Duration. It has a method that can calculate the duration between 2 LocalTime instances. Alternatively, we can also create our own Duration class, but I don't think that is very necessary.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (tyanhan) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:160]: For now, it is set to only accept hours. I will be updating this later on.

**Pull request closed without merging**